EIPS/eip-2680.md

```diff
@@ -0,0 +1,130 @@
+
+## Specification
+
+There are four elements for a wallet that need to be addressed.  These are defined below.
+
+### Base location
+The base location is required to be well-known, either pre-defined or defined by the storage system's connection parameters.
+
+For filesystems the pre-defined base location for different operating systems is as follows:
+
+  - Windows: `%APPDATA%\ethereum2\wallets`
+  - MacOSX: `${HOME}/Library/Application Support/ethereum2/wallets`
+  - Linux: `${HOME}/.config/ethereum2/wallets`
+
+For other hierarchical stores, for example Amazon S3, the base location MUST be the lower-case hex string representing the [SHA-256](https://web.archive.org/web/20130526224224/http://csrc.nist.gov/groups/STM/cavp/documents/shs/sha256-384-512.pdf) hash of the string "Ethereum 2 wallet:" appended with the identifier for the hierarchical store.  For example, if the account ID for a user's Amazon S3 account is "AbC0438EB" then:
+
+  - string would be `Ethereum 2 wallet:AbC0438EB`
+  - SHA-256 hash of string would be the byte array `0x991ec14a8d13836b10d8c3039c9e30876491cb8aa9c9c16967578afc815c9229`
+  - base location would be the string `991ec14a8d13836b10d8c3039c9e30876491cb8aa9c9c16967578afc815c9229`
+
+For non-hierarchical stores there is no base location.
+
+### Wallet container
+The wallet container holds the walletstore and related keystores.
+
+The wallet container is identified by the wallet's UUID.  It MUST be a string following the syntactic structure as laid out in [section 3 of RFC 4122](https://tools.ietf.org/html/rfc4122#section-3).
+
+### Walletstore
+The walletstore element contains the walletstore and is held within the wallet container.  It is identified by the wallet's UUID.  It MUST be a string following the syntactic structure as laid out in [section 3 of RFC 4122](https://tools.ietf.org/html/rfc4122#section-3).
+
+### Keystore
+The keystore element contains the keystore for a given key and is held within the wallet container.  It is identified by the key's UUID.  It MUST be a string following the syntactic structure as laid out in [section 3 of RFC 4122](https://tools.ietf.org/html/rfc4122#section-3).
+
+## Hierarchical store example
+Hierarchical stores are a common way to store and organize information.  The most common example is the filesystem, but a number of object-based stores such as Amazon S3 also provide hierarchical naming.
+
+Putting these elements together for a sample wallet with wallet UUID `1f031fff-c51d-44fc-8baf-d6b304cb70a7` and key UUIDs `1302106c-8441-4e2e-b687-6c77f49fc624` and `4a320100-83fd-4db7-8126-6d6d205ba834` gives the following layout:
+
+```
+- 1f031fff-c51d-44fc-8baf-d6b304cb70a7
++- 1302106c-8441-4e2e-b687-6c77f49fc624
++- 1f031fff-c51d-44fc-8baf-d6b304cb70a7
++- 4a320100-83fd-4db7-8126-6d6d205ba834
+```
+
+## Non-hierarchical store example
```

Comment:
This and the following sections should be in one of the template sections.  My guess is you just forgot a `#` at the start?
```suggestion
### Non-hierarchical store example
```